src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/Pipeline.cs

```diff
@@ -0,0 +1,539 @@
+        /// <feature>http://tizen.org/feature/machine_learning.inference</feature>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to the wrong pipeline description or internal error.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public Pipeline(string description)
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            if (string.IsNullOrEmpty(description))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The pipeline description is invalid");
+
+            _stateChangedCallback = (state, _) =>
+            {
+                _stateChanged?.Invoke(this, new StateChangedEventArgs(state));
+            };
+
+            NNStreamerError ret = Interop.Pipeline.Construct(description, _stateChangedCallback, IntPtr.Zero, out _handle);
+            NNStreamer.CheckException(ret, "Failed to create Pipeline instance");
+
+            /* Init node list */
+            _sinkNodeList = new Dictionary<string, SinkNode>();
+        }
+
+        /// <summary>
+        /// Destructor of the Pipeline instance.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        ~Pipeline()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Internal method to get the native handle of pipeline.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        internal IntPtr GetHandle()
+        {
+            return _handle;
+        }
+
+        /// <summary>
+        /// The state of pipeline.
+        /// </summary>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed to get the pipeline state.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public PipelineState State
+        {
+            get
+            {
+                NNStreamer.CheckNNStreamerSupport();
+
+                int state = 0;
+                NNStreamerError ret = NNStreamerError.None;
+
+                ret = Interop.Pipeline.GetState(_handle, out state);
+                if (ret == NNStreamerError.None && state == 0)
+                    ret = NNStreamerError.InvalidOperation;
+
+                NNStreamer.CheckException(ret, "Failed to get the pipeline state because of internal error");
+
+                return (PipelineState) state;
+            }
+        }
+
+        /// <summary>
+        /// Event to be invoked when the pipeline state is changed.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public event EventHandler<StateChangedEventArgs> StateChanged
```

Comment:
You can omit it, if no additional implementation
``` c#
public event EventHandler<StateChangedEventArgs> StateChanged;
```
And you can use `StateChanged` instead of `_stateChanged`

Reply:
updated